### ISSUE TITLE
Can't swap! a cursor before derefing it

### ISSUE THREAD

**Issue by icambron** [reagent-project/reagent#159]
In 0.5.0-alpha3 (edit: or 0.5.0):

```clj
cljs.user=> (require '[reagent.core :as r])
nil
cljs.user=> (def my-atom (r/atom {:foo {:bar "baz" :quux "zob"}}))
#<Atom: {:foo {:bar "baz", :quux "zob"}}>
nil
cljs.user=> (do (def my-cursor (r/cursor my-atom [:foo])) nil) ;;nil to stop printing from derefing
nil
cljs.user=> (swap! my-cursor assoc :bar "yaks")
{:bar "yaks”}
```

Notice that `:quux "zob"` has disappeared. If I deref the cursor before the swap, everything works as expected.

**Comment by JulianLeviston** [comment]
Why are you using 0.5.0 alpha3 and not 0.5.0 final?

**Comment by icambron** [comment]
@JulianLeviston I didn't see a tag for it in this repo and didn't know it existed. Happy to test to see if this issue is present there.

**Comment by icambron** [comment]
Same issue with 0.5.0. Updated the description to reflect that.

**Comment by zarkone** [comment]
can't reproduce, `[reagent "0.5.0"]`

```
cljs.user=> (require '[reagent.core :as r])
nil
cljs.user=> (def my-atom (r/atom {:foo {:bar "baz" :quux "zob"}}))
#<Atom: {:foo {:bar "baz", :quux "zob"}}>
cljs.user=> (do (def my-cursor (r/cursor my-atom [:foo])) nil)
nil
cljs.user=> (swap! my-cursor assoc :bar "yaks")
{:bar "yaks", :quux "zob"}
```

**Comment by icambron** [comment]
Thanks for doing that - it prompted me to retry after being more thorough in making sure I was really running the version of Reagent that I thought I was. After doing so, I can confirm the issue was fixed in `0.5.0`. Closing this out.
